src/cmd/create.rs

```diff
@@ -116,24 +133,33 @@ impl Sharded {
             None => None,
         };
         let password = get_password(true)?;
-        let tag = KeyTag {
-            network: self.network,
-            key_type: self.key_type,
-        };
+
         let shard_config = ShardConfig {
             key_share_count: self.key_share_count,
             recovery_threshold: self.recovery_threshold,
         };
 
-        let wallet = Wallet::builder()
+        let mut builder = Wallet::builder()
             .output(&self.output)
             .password(&password)
-            .key_tag(&tag)
             .force(self.force)
-            .seed_type(self.seed.to_owned())
-            .seed_words(seed_words)
-            .shard(Some(shard_config))
-            .create()?;
+            .shard(Some(shard_config));
+
+        builder = match (self.network, self.key_type, &self.swarm) {
+            (_, _, Some(swarm)) => builder.from_swarm(swarm.to_path_buf()),
+            (network, key_type, None) => {
+                let tag = KeyTag {
+                    network: network.unwrap_or(Network::MainNet),
```

Comment:
ditto.. more generally you're repeating yourself here, perhaps this could be pulled out into a function?